### DISCUSSION TITLE
Implementation of unsupervised GNN

### DISCUSSION THREAD

**Discussion by jianchaoji** [pyg-team/pytorch_geometric#2212] [Q&A]
Do we have other implementation of unsupervised GNN model, other than the given example :graph_sage_unsup.py
Such as unsupervised GCN and GAT

**Comment by rusty1s** [comment]
We have a few more:
* `examples/node2vec.py`
* `examples/autoencoder.py`
* `examples/link_pred.py`
* `examples/infomax.py`

**Reply by jianchaoji** [reply]
Thank you for your reply! Is there any plan to update unsupervised GCN and GAT ? If no can I send you a PR when I implement my unsupervised GCN and GAT examples?

**Reply by rusty1s** [reply]
I'm not exactly sure what you mean with unsupervised GCN. For example, the unsupervised `autoencoder.py` example already utilizes GCN, and the `infomax.py` examples utilizes GAT.

**Reply by jianchaoji** [reply]
Oh,I didn't recogenize that since the loss function is different from `graph_sage_unsup`. Thanks for your explanation！
